### PR TITLE
Administration communities

### PR DESCRIPTION
**Pull request by kpsherva**
* closes https://github.com/inveniosoftware/invenio-communities/issues/821

Screenshot 2022-11-03 at 16 11 00
Screenshot 2022-11-03 at 16 11 09
Screenshot 2022-11-03 at 16 11 16

### CODE REVIEW

File: invenio_communities/assets/semantic-ui/js/invenio_communities/administration/FeatureModal.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    const { resource } = this.props;
+    try {
+      this.setState({ loading: true });
+      const cancellableFetch = withCancel(
+        InvenioAdministrationCommunitiesApi.getFeatured(resource.links.featured)
+      );
+      const featuredList = await cancellableFetch.promise;
+      this.setState({ loading: false, featuredList: featuredList.data });
+    } catch (e) {
+      console.error(e);
```
Comment (jrcastro2) [inline invenio_communities/assets/semantic-ui/js/invenio_communities/administration/FeatureModal.js:31]: Minor: Missing error handling

File: invenio_communities/assets/semantic-ui/js/invenio_communities/administration/featured.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  fetchData = async () => {
+    const { data } = this.props;
+    try {
+      this.setState({ loading: true });
+      const cancellableFetch = withCancel(
+        InvenioAdministrationCommunitiesApi.getFeatured(data.links.featured)
+      );
+      const featuredList = await cancellableFetch.promise;
+      this.setState({ loading: false, featuredList: featuredList.data });
+      console.log(featuredList);
```
Comment (jrcastro2) [inline invenio_communities/assets/semantic-ui/js/invenio_communities/administration/featured.js:35]:
```suggestion
```

File: invenio_communities/communities/schema.py [hunk 3]
```diff
@@ -7,6 +7,7 @@
 # under the terms of the MIT License; see LICENSE file for more details.
 
 """Community schema."""
+import typing
```
Comment (jrcastro2) [inline invenio_communities/communities/schema.py:10]: Leftover?
```suggestion
```

File: invenio_communities/assets/semantic-ui/js/invenio_communities/administration/details.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+const uiSchema = JSON.parse(domContainer.dataset?.uiConfig);
+const requestHeaders = JSON.parse(domContainer.dataset?.requestHeaders);
+
+const createdBySystem = (data) => data?.system_created;
+
+const overridenComponents = {
+  EditAction: parametrize(Edit, {
+    disable: createdBySystem,
+    disabledMessage:
+      "This set is not editable as it was created by the system.",
```
Comment (jrcastro2) [inline invenio_communities/assets/semantic-ui/js/invenio_communities/administration/details.js:36]:
```suggestion
      i18next.t("This set is not editable as it was created by the system."),
```

File: invenio_communities/assets/semantic-ui/js/invenio_communities/administration/details.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+  }),
+  DeleteAction: parametrize(Delete, {
+    disable: createdBySystem,
+    disabledMessage:
+      "This set is not deletable as it was created by the system.",
```
Comment (jrcastro2) [inline invenio_communities/assets/semantic-ui/js/invenio_communities/administration/details.js:41]:
```suggestion
      i18next.t("This set is not deletable as it was created by the system."),
```

**Review by jrcastro2 — COMMENTED** [review]
LGTM! Just a couple of translations are missing.

File: invenio_communities/assets/semantic-ui/js/invenio_communities/administration/details.js (hunk 5 above)
Comment (kpsherva) [reply invenio_communities/assets/semantic-ui/js/invenio_communities/administration/details.js:41]: nice catch! thanks

File: invenio_communities/assets/semantic-ui/js/invenio_communities/administration/config.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+  withCredentials: true,
+  xsrfCookieName: "csrftoken",
+  xsrfHeaderName: "X-CSRFToken",
+  headers: {
+    "Accept": "application/vnd.inveniordm.v1+json",
+    "Content-Type": "application/json",
+  },
+};
+
+export const http = axios.create(apiConfig);
```
Comment (zzacharo) [inline invenio_communities/assets/semantic-ui/js/invenio_communities/administration/config.js:19]: Is this suppose to use the central axios from react-invenio-forms?

File: invenio_communities/assets/semantic-ui/js/invenio_communities/administration/config.js [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+/**
+ * Wrap a promise to be cancellable and avoid potential memory leaks
+ * https://reactjs.org/blog/2015/12/16/ismounted-antipattern.html
+ * @param promise the promise to wrap
+ * @returns {Object} an object containing the promise to resolve and a `cancel` fn to reject the promise
+ */
+export const withCancel = (promise) => {
```
Comment (zzacharo) [inline invenio_communities/assets/semantic-ui/js/invenio_communities/administration/config.js:27]: leftover?

**Comment by jennur** [comment]
### Minor comments:
Screenshot 2022-11-14 at 14 01 37

Maybe we should add the name of the community in the "Feature" header? Just to not cause any insecurities at all about what the user is about to do.

Screenshot 2022-11-14 at 14 01 53
Center the star and button in the cell? 🙃

File: invenio_communities/assets/semantic-ui/js/invenio_communities/administration/FeatureModal.js [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+    return (
+      <Modal role="dialog" open={modalOpen}>
+        <Modal.Content>
+          <Header as="h2"> Planned features</Header>
+          {loading && <Loader active={loading} />}
+          {error && <ErrorMessage error={error} />}
+          {!loading && !error && (
+              <Table celled striped collapsing>
+                <Table.Header>
+                  <Table.HeaderCell>Start date</Table.HeaderCell>
```
Comment (jennur) [inline invenio_communities/assets/semantic-ui/js/invenio_communities/administration/FeatureModal.js:55]:
```suggestion
                  <Table.HeaderCell>{i18next.t("Start date")}</Table.HeaderCell>
```

File: invenio_communities/assets/semantic-ui/js/invenio_communities/administration/FeatureModal.js [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+                  <Table.HeaderCell>Active</Table.HeaderCell>
```
Comment (jennur) [inline invenio_communities/assets/semantic-ui/js/invenio_communities/administration/FeatureModal.js:56]:
```suggestion
                  <Table.HeaderCell>{i18next.t("Active")}</Table.HeaderCell>
```

File: invenio_communities/assets/semantic-ui/js/invenio_communities/administration/FeatureModal.js [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+
+  render() {
+    const { featuredList, loading, error } = this.state;
+    const { children, modalOpen } = this.props;
+    const now = DateTime.now();
+
+    return (
+      <Modal role="dialog" open={modalOpen}>
+        <Modal.Content>
+          <Header as="h2"> Planned features</Header>
```
Comment (jennur) [inline invenio_communities/assets/semantic-ui/js/invenio_communities/administration/FeatureModal.js:49]:
```suggestion
          <Header as="h2">{i18next.t("Planned features")}</Header>
```

File: invenio_communities/assets/semantic-ui/js/invenio_communities/administration/featured.js [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+  }
+
+  render() {
+    const { featuredList, loading, error } = this.state;
+    const { data } = this.props;
+    const now = DateTime.now();
+
+    return (
+      <>
+        <Header as="h2"> Planned features</Header>
```
Comment (jennur) [inline invenio_communities/assets/semantic-ui/js/invenio_communities/administration/featured.js:53]:
```suggestion
        <Header as="h2">{i18next.t("Planned features")}</Header>
```

File: invenio_communities/assets/semantic-ui/js/invenio_communities/administration/FeatureModal.js [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+                    return (
+                      <Table.Row>
+                        <Table.Cell
+                          key={row.id}
+                          data-label={row.id}
+                          className="word-break-all"
+                        >
+                          <DateFormatter value={row.start_date} />
+                        </Table.Cell>
+                        <Table.Cell>
```
Comment (jennur) [inline invenio_communities/assets/semantic-ui/js/invenio_communities/administration/FeatureModal.js:70]:
```suggestion
                        <Table.Cell textAlign="center">
```
?

File: invenio_communities/assets/semantic-ui/js/invenio_communities/administration/featured.js [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+                return (
+                  <Table.Row>
+                    <Table.Cell
+                      key={row.id}
+                      data-label={row.id}
+                      className="word-break-all"
+                    >
+                      <DateFormatter value={row.start_date} />
+                    </Table.Cell>
+                    <Table.Cell>
```
Comment (jennur) [inline invenio_communities/assets/semantic-ui/js/invenio_communities/administration/featured.js:78]:
```suggestion
                    <Table.Cell textAlign="center">
```

File: invenio_communities/assets/semantic-ui/js/invenio_communities/administration/featured.js [hunk 14]
```diff
@@ -0,0 +1,107 @@
...
+                      <BoolFormatter
+                        icon="star"
+                        color="yellow"
+                        value={startDate <= now}
+                      />
+                    </Table.Cell>
+                    <Table.Cell>
```
Comment (jennur) [inline invenio_communities/assets/semantic-ui/js/invenio_communities/administration/featured.js:85]:
```suggestion
                    <Table.Cell textAlign="center">
```

**Review by jennur — APPROVED** [review]
Just translations and text alignment suggestions!

File: invenio_communities/assets/semantic-ui/js/invenio_communities/administration/config.js (hunk 6 above)
Comment (kpsherva) [reply invenio_communities/assets/semantic-ui/js/invenio_communities/administration/config.js:19]: not integrated yet, therefore I didn't use it

File: invenio_communities/administration/__init__.py [hunk 15]
```diff
@@ -0,0 +1,7 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2022 CERN
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
```
Comment (slint) [inline invenio_communities/administration/__init__.py:5]: Minor: check copyright headers (also on other files, I saw some `invenio-administration` originated headers)

File: invenio_communities/communities/services/results.py [hunk 16]
```diff
@@ -57,46 +90,39 @@ def total(self):
...
+        """Get a dictionary for the request."""
+        res = self.data
+        if self._errors:
+            res["errors"] = self._errors
         return res
 
 
-class CommunityFeaturedList(BaseServiceListResult):
-    """List of featured community entry result items."""
+class FeaturedCommunityItem(CommunityItem):
```
Comment (slint) [inline invenio_communities/communities/services/results.py:127]:
```suggestion
class CommunityFeaturedItem(CommunityItem):
```

minor: for consistency, though I highly dislike this name ... on the other hand, changing `CommunityFeaturedList` to `FeaturedCommunityList` is backward incompatible (but doable via an alias + deprecation warning).

**Review by slint — COMMENTED** [review]
LGTM! A possible rename for consistency, and some copyright headers.

**Review by jrcastro2 — APPROVED** [review]

**Pull request merged**